app/views/devise/registrations/_form.html.erb

```diff
@@ -1,44 +1,44 @@
 <div class="card-main">
-  <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { class: 'form' }) do |form| %>
-
-    <div class="form__section">
-      <span class="fas fa-user form__icon"></span>
-      <%= form.text_field :username, required: true, autofocus: true, placeholder: 'Username', class: 'form__element form__element--with-icon', data: {test_id: 'username_field'} %>
+  <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { class: 'form', 'data-controller':"signup-validation", 'data-action':"submit->signup-validation#validateAll" }) do |form| %>
+
+    <div class="form-field mb-4">
+      <div class="form__section mb-0">
+        <span class="fas fa-user form__icon"></span>
+        <%= form.text_field :username, 'data-action':"blur->signup-validation#validate", 'data-attr':"signup-validation.username", required: true, placeholder: 'Username', class: 'form__element form__element--with-icon', data: {test_id: 'username_field'} %>
```

Comment:
We can add any data attributes to the ruby data hash to cut down on a little bit of code

```suggestion
        <%= form.text_field(
          :username, 
          required: true, 
          placeholder: 'Username', 
          class: 'form__element form__element--with-icon', 
          data: { test_id: 'username_field', action: "blur->signup-validation#validate", attr: "signup-validation.username"} 
        )%>
```

Can we do this for all the form fields with validation data attributes?